samples/bluetooth/tmap_bmr/src/bap_broadcast_sink.c

```diff
@@ -0,0 +1,414 @@
+		return false;
+	}
+
+	for (size_t i = 0U; i < base.subgroup_count; i++) {
+		for (size_t j = 0U; j < base.subgroups[i].bis_count; j++) {
+			const uint8_t index = base.subgroups[i].bis_data[j].index;
+
+			base_bis_index_bitfield |= BIT(index);
+		}
+	}
+
+	bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
+	k_sem_give(&sem_base_received);
+
+	return false;
+}
+
+static void broadcast_pa_recv(struct bt_le_per_adv_sync *sync,
+			      const struct bt_le_per_adv_sync_recv_info *info,
+			      struct net_buf_simple *buf)
+{
+	bt_data_parse(buf, pa_decode_base, NULL);
+}
+
+static void broadcast_pa_synced(struct bt_le_per_adv_sync *sync,
+				struct bt_le_per_adv_sync_synced_info *info)
+{
+	k_sem_give(&sem_pa_synced);
+}
+
+static void pa_synced_cb(struct bt_bap_broadcast_sink *sink,
+			 struct bt_le_per_adv_sync *sync,
+			 uint32_t broadcast_id)
+{
+	if (broadcast_sink != NULL) {
+		printk("Unexpected PA sync\n");
+		return;
+	}
+
+	printk("PA synced for broadcast sink %p with broadcast ID 0x%06X\n",
+	       sink, broadcast_id);
+
+	broadcast_sink = sink;
+
+	k_sem_give(&sem_sink_created);
+}
+
+static void syncable_cb(struct bt_bap_broadcast_sink *sink, bool encrypted)
+{
+	k_sem_give(&sem_syncable);
+}
+
+static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap_base *base)
+{
+	k_sem_give(&sem_base_received);
+}
+
+static void pa_sync_lost_cb(struct bt_bap_broadcast_sink *sink)
+{
+	if (broadcast_sink == NULL) {
+		printk("Unexpected PA sync lost\n");
+		return;
+	}
+
+	printk("Sink %p disconnected\n", sink);
+	broadcast_sink = NULL;
+	k_sem_give(&sem_pa_sync_lost);
+}
+
+static struct bt_bap_broadcast_sink_cb broadcast_sink_cbs = {
+	.pa_synced = pa_synced_cb,
+	.syncable = syncable_cb,
+	.base_recv = base_recv_cb,
+	.pa_sync_lost = pa_sync_lost_cb
+};
+
+static int reset(void)
+{
+	int err;
+
+	if (broadcast_sink != NULL) {
+		err = bt_bap_broadcast_sink_delete(broadcast_sink);
+		if (err) {
+			printk("Deleting broadcast sink failed (err %d)\n", err);
+
+			return err;
+		}
+
+		broadcast_sink = NULL;
+	}
+	k_sem_reset(&sem_pa_synced);
+	k_sem_reset(&sem_base_received);
+	k_sem_reset(&sem_sink_created);
+	k_sem_reset(&sem_syncable);
+	k_sem_reset(&sem_pa_sync_lost);
+
+	return 0;
+}
+
+int bap_broadcast_sink_init(void)
+{
+	int err;
+
+	bt_bap_broadcast_sink_register_cb(&broadcast_sink_cbs);
+	bt_le_per_adv_sync_cb_register(&broadcast_sync_cb);
+
+	err = bt_pacs_cap_register(BT_AUDIO_DIR_SINK, &cap);
+	if (err) {
+		printk("Capability register failed (err %d)\n", err);
+		return err;
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(streams); i++) {
+		streams[i].ops = &stream_ops;
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(streams_p); i++) {
+		streams_p[i] = &streams[i];
+	}
+
+	return 0;
+}
+
+int bap_broadcast_sink_run(void)
+{
+	int err;
+
+	while (true) {
+		err = reset();
+		if (err != 0) {
+			printk("Resetting failed: %d - Aborting\n", err);
+			return err;
```

Comment:
Either declare `err` inside the while loop, or use `break` instead of `return` in the loop